src/bot.js

```diff
@@ -1,38 +1,186 @@
 /**
- * This bot is supposed to send a NASA photo every day in a specific channel
+ * This bot sends the NASA APOD in discord channel on configurable time every day
  */
-const djs = require('discord.js');
-const schedule = require('node-schedule');
-const { CHANNEL, TOKENS } = require('./constants.js');
-const getAPOD = require('./nasa.js');
-const { onMessage, onReady } = require('./events.js');
+const https = require('https');
+const ns = require('node-schedule');
+const discordjs = require('discord.js');
+const client = new discordjs.Client();
+const { TOKENS, NASA_URL } = require('./constants.js');
 
 /**
- * This is where everything starts
+ * Reaches out to the NASA_URL and returns whatever is needed
+ * @returns {Promise<NASAData>}
  */
-async function main() {
-  const client = new djs.Client();
-  
-  client.on('ready', onReady.bind(client));
-  client.on('message', onMessage.bind(client));
+async function getAPOD() {
+  return new Promise((res, rej) => {
+    let result = '';
+    const req = https.get(NASA_URL, (response) => {
+      // Listen for incoming data on the stream
+      response.on('data', data => result += data);
+      // Handle errors on the stream and call the Promise's reject function. 
+      // This let's users of this function know that something went wrong
+      response.on('error', rej);
+      // When the stream has finished sending data let's send it back to the
+      // original function all
+      response.on('end', _ => {
+        // deserialize the data (turn a buffer into object data that our code
+        // can understand)
+        if (result.length === 0) {
+          const msg = "Result was empty and status code was: "
+            + response.statusCode;
+          rej(new Error(msg));
+        }
+        const deserialized = JSON.parse(result);
+        // call the Promise's resolve callback function which let the user of
+        // the function know that this is what they want
+        res(deserialized);
+      });
+    });
+  });
+}
 
-  await client.login(TOKENS.discord);
+/**
+ * @param {string} jobChannel
+ * This function sends the APOD as a formatted embed Discord message
+ */
+async function sendAPOD(jobChannel) {
+  const data = await getAPOD();
+  const channel = await client.channels.fetch(jobChannel, true);
+  console.debug(data);
```

Comment:
nit: No context for this debug log, although this is the only debug log, it can stack up and you'll end up not knowing what you're looking at or what users of this project are looking at. Perhaps don't touch this now until you get a proper logging library like [log4js](https://www.npmjs.com/package/log4js)

Reply:
e207e731107ecbf7334dc591dc24ed274e61ea86